### PR TITLE
[backend] chore: move xtm hub file and add api url config

### PR DESCRIPTION
**Pull request by Kroustille**
### Proposed changes

* clean old XTM Hub files
* add required API URL override for future features

### Testing Instructions

1. Add an override API Url in config
2. Make sure `XTMHubConfig.getApiUrl()` returns the overriden configuration

### Related issues

* Related to #4022

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality
- [ ] For bug fix -> I implemented a test that covers the bug

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/xtmhub/config/XTMHubConfig.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  @NotNull private Boolean enable;
+
+  @JsonProperty("url")
+  private String url;
+
+  @JsonProperty("override_api_url")
+  private String override_api_url;
+
+  public String getApiUrl() {
+    if (this.override_api_url != null && this.override_api_url.length() > 0) {
```
Comment (antoinemzs) [inline openbas-api/src/main/java/io/openbas/xtmhub/config/XTMHubConfig.java]: Prefer the static `StringUtils.isBlank` from apache commons for blank checking

File: openbas-api/src/test/java/io/openbas/config/XTMHubConfigTest.java [hunk 2]
```diff
@@ -27,6 +27,21 @@ public class withEnabledXTMHub {
...
+  @DisplayName("When XTM Hub API URL is overridden")
+  public class withOverrideApiUrl {
+
+    @Autowired private XTMHubConfig xtmHubConfig;
+
+    @Test
+    @DisplayName("returns overridden API URL")
+    public void shouldReturnOverriddenApiUrl() {
+      assertThat(xtmHubConfig.getApiUrl()).isEqualTo("http://localhost:4002");
     }
```
Comment (antoinemzs) [inline openbas-api/src/test/java/io/openbas/config/XTMHubConfigTest.java:45]: please add a test that ensures that the primary url is used when override is null/blank/empty

Comment (Kroustille) [reply openbas-api/src/test/java/io/openbas/config/XTMHubConfigTest.java:45]: It's kinda done in a test above (`shouldReturnEnabledStatusAndUrl`) where I check for `getApiUrl()` value with only `url` specified. 

Do you wish to have an additional test ?

**Comment by Kroustille** [comment]
> Please add a commented-out key for `openbas.xtm.hub.override_api_url` in application.properties with no example value.
> 
> Please document the use of this key with a PR in the OpenAEV-Platform/docs repository

It's done for the configuration example in `application.properties`. [Here](https://github.com/OpenAEV-Platform/docs/pull/199) you can find the pull request to update documentation 😁

File: openbas-api/src/test/java/io/openbas/config/XTMHubConfigTest.java (hunk 2 above)
Comment (antoinemzs) [reply openbas-api/src/test/java/io/openbas/config/XTMHubConfigTest.java:45]: Ah yes good call, no that's all good, sorry!
